src/config.rs

```diff
@@ -510,26 +559,74 @@ impl Cfg {
         Ok(None)
     }
 
+    fn parse_override_file<S: AsRef<str>>(contents: S) -> Result<Option<OverrideFile>> {
+        let contents = contents.as_ref();
+
+        if contents.lines().any(|line| line.starts_with("[toolchain]")) {
+            toml::from_str::<OverrideFile>(contents)
+                .map(|file| Some(file))
+                .map_err(|e| ErrorKind::ParsingOverride(e).into())
```

Comment:
If `channel` is made optional, here is where you'd need to verify that the constructed `OverrideFile` has at least one of `channel`, `components`, or `targets` specified.